tests/std/tests/P1522R1_difference_type/test.cpp

```diff
@@ -35,9 +35,133 @@ namespace ordtest {
 
 #define STATIC_ASSERT(...) static_assert(__VA_ARGS__, #__VA_ARGS__)
 
+#if _HAS_CXX20
+#define CONSTEVAL consteval
+#else // ^^^ _HAS_CXX20 / !_HAS_CXX20 vvv
+#define CONSTEVAL constexpr
+#endif // ^^^ !_HAS_CXX20 ^^^
+
+#if _HAS_CXX20 && !defined(__clang__) // TRANSITION, LLVM-51840
+#define CONSTEVAL_CLANG_WORKAROUND consteval
+#else // ^^^ _HAS_CXX20 && !defined(__clang__) / !_HAS_CXX20 || defined(__clang__) vvv
+#define CONSTEVAL_CLANG_WORKAROUND constexpr
+#endif // ^^^ !_HAS_CXX20 || defined(__clang__) ^^^
+
+#if _HAS_CXX17
+#define NODISCARD [[nodiscard]]

```

Comment:
We use `[[nodiscard]]` unconditionally in test and product code, we should do so here as well.